### PR TITLE
Riak backend for private storage

### PR DESCRIPTION
**Pull request by michalwski**

### CODE REVIEW

**Comment by mongoose-im** [comment]
travis is using test branch `riak-private` from https://github.com/esl/ejabberd_tests/tree/riak-private

**Comment by mongoose-im** [comment]
travis is using test branch `riak-private` from https://github.com/esl/ejabberd_tests/tree/riak-private

**Comment by mongoose-im** [comment]
travis is using test branch `riak-private` from https://github.com/esl/ejabberd_tests/tree/riak-private

File: apps/ejabberd/src/mongoose_riak.erl [hunk 1]
```diff
@@ -30,9 +30,11 @@
 -export([update_type/3, update_type/4]).
 -export([fetch_type/2, fetch_type/3]).
 -export([list_keys/1]).
+-export([list_buckets/1]).
```
Comment (ppikula) [inline apps/ejabberd/src/mongoose_riak.erl]: is list_buckets used somewhere? Shall I remove it?

Comment (michalwski) [inline apps/ejabberd/src/mongoose_riak.erl]: It's a helper function I used while developing. I left it because I found it useful.

**Pull request merged**